relay-server/src/service.rs

```diff
@@ -133,6 +133,7 @@ impl ServiceState {
 
         let global_config =
             GlobalConfigService::new(config.clone(), upstream_relay.clone()).start();
+        let global_config_subscription = global_config.send(Subscribe).await?;
```

Comment:
This way you expose the internal of the global config, which does not seem like a good idea. Also, it's already starting in the. correct order, first `GlobalConfigService` and then `EnveloperProcessortService` which subscribes to the `GlobalConfigService`, which already should cover the dependency you're talking about.